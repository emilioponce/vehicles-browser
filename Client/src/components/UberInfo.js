import React from 'react'
import PropTypes from 'prop-types'

import styles from './UberInfo.module.css'

const UberInfo = ({
  title,
  position,
  address,
  fuel,
  engineType,
  condition,
  name
}) => {
  return (
    <div className={styles.uberInfo}>
      <div className={styles.leftSide}>
        <div className={styles.title}>{title}</div>
        <div>Address: {address}</div>
        <div>Position:</div>
        <div className={styles.coordinates}>
          <div>Latitude: {position[1].toFixed(4)}</div>
          <div> Longitude: {position[0].toFixed(4)}</div>
        </div>
      </div>
      <div className={styles.rightSide}>
        <div>Engine: {engineType}</div>
        <div>Fuel : {fuel}</div>
        <div>{condition}</div>
        <div>Name: {name}</div>
      </div>
    </div>
  )
}

UberInfo.propTypes = {
  title: PropTypes.string.isRequired,
  position: PropTypes.array.isRequired,
  address: PropTypes.string.isRequired,
  fuel: PropTypes.number.isRequired,
  engineType: PropTypes.string.isRequired,
  condition: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired
}

export default UberInfo
