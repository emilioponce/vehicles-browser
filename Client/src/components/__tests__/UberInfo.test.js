import React from 'react'
import ReactDOM from 'react-dom'
import UberInfo from '../UberInfo'
import renderer from 'react-test-renderer'

const component = (
  <UberInfo
    title="UBER"
    position={[41.5532, 2.0088]}
    address="Address 1 Barcelona"
    fuel={34}
    engineType="CE"
    condition="Exterior: UNACCEPTABLE | Interior:GOOD"
    name="1234-ABC"
  />
)

describe('Component UberInfo', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(component, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('snapshot renders correctly', () => {
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
