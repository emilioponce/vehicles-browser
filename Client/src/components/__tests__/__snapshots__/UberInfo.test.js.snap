// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component UberInfo snapshot renders correctly 1`] = `
<div
  className="uberInfo"
>
  <div
    className="leftSide"
  >
    <div
      className="title"
    >
      UBER
    </div>
    <div>
      Address: 
      Address 1 Barcelona
    </div>
    <div>
      Position:
    </div>
    <div
      className="coordinates"
    >
      <div>
        Latitude: 
        2.0088
      </div>
      <div>
         Longitude: 
        41.5532
      </div>
    </div>
  </div>
  <div
    className="rightSide"
  >
    <div>
      Engine: 
      CE
    </div>
    <div>
      Fuel : 
      34
    </div>
    <div>
      Exterior: UNACCEPTABLE | Interior:GOOD
    </div>
    <div>
      Name: 
      1234-ABC
    </div>
  </div>
</div>
`;
